osaca/parser/parser_x86intel.py

```diff
@@ -258,32 +258,24 @@ def process_operand(self, operand):
         return operand
 
     def process_register(self, operand):
-        return RegisterOperand(
-            name=operand["name"],
-        )
+        return RegisterOperand(name=operand.name)
 
     def process_memory_address(self, memory_address):
         """Post-process memory address operand"""
         # TODO: Use the ptr type.
-        ptr_type = memory_address["ptr_type"]
-        register_expression = memory_address["register_expression"]
+        ptr_type = memory_address.ptr_type
+        register_expression = memory_address.register_expression
         displacement = register_expression.get(
             "displacement1",
-            register_expression.get("displacement2", None)
+            register_expression.get("displacement2")
         )
-        base = register_expression.get("base", None)
-        index = register_expression.get("index", None)
-        scale = 1 if "scale" not in register_expression else int(register_expression["scale"], 0)
-        displacementOp = None
-        baseOp = None
-        indexOp = None
-        if displacement:
-            displacementOp = self.process_immediate(displacement["immediate"])
-        if base:
-            baseOp = RegisterOperand(name=base["name"])
-        if index:
-            indexOp = RegisterOperand(name=index["name"])
-        new_dict = MemoryOperand(offset=displacementOp, base=baseOp, index=indexOp, scale=scale)
+        base = register_expression.get("base")
+        index = register_expression.get("index")
+        scale = int(register_expression.get("scale", "1"), 0)
+        displacement_op = self.process_immediate(displacement.immediate) if displacement else None
+        base_op = RegisterOperand(name=base.name) if base else None
+        index_op = RegisterOperand(name=index.name) if index else None
+        new_dict = MemoryOperand(offset=displacement_op, base=base_op, index=index_op, scale=scale)
```

Comment:
new_dict is a weird name for a thing that is not a dict (it is, well, a `MemoryOperand`, which is an `Operand`, which is not a dict).